src/setup_node_env/node_version_validator.js

```diff
@@ -37,18 +37,31 @@ var rawRequiredVersion = (pkg && pkg.engines && pkg.engines.node) || null;
 var requiredVersion = rawRequiredVersion ? 'v' + rawRequiredVersion : rawRequiredVersion;
 var currentVersionMajorMinorPatch = currentVersion.match(/^v(\d+)\.(\d+)\.(\d+)/);
 var requiredVersionMajorMinorPatch = requiredVersion.match(/^v(\d+)\.(\d+)\.(\d+)/);
+var currentMinorPatch = currentVersionMajorMinorPatch[2].concat(
+  '.',
+  currentVersionMajorMinorPatch[3]
+);
+var requiredMinorPatch = requiredVersionMajorMinorPatch[2].concat(
+  '.',
+  requiredVersionMajorMinorPatch[3]
+);
 var isVersionValid =
   currentVersionMajorMinorPatch[1] === requiredVersionMajorMinorPatch[1] &&
-  currentVersionMajorMinorPatch[2] === requiredVersionMajorMinorPatch[2] &&
-  parseInt(currentVersionMajorMinorPatch[3], 10) >= parseInt(requiredVersionMajorMinorPatch[3], 10);
+  parseFloat(currentMinorPatch) >= parseFloat(requiredMinorPatch);
+var isMinorValid = currentVersionMajorMinorPatch[2] === requiredVersionMajorMinorPatch[2];
 
 // Validates current the NodeJS version compatibility when OpenSearch Dashboards starts.
 if (!isVersionValid) {
   var errorMessage =
     `OpenSearch Dashboards was built with ${requiredVersion} and does not support the current Node.js version ${currentVersion}. ` +
     `Please use Node.js ${requiredVersion} or a higher patch version.`;
-
   // Actions to apply when validation fails: error report + exit.
   console.error(errorMessage);
   process.exit(1);
+} else if (!isMinorValid) {
```

Comment:
With my proposal, this should be
```suggestion
} else if (isMinorValid) {
```

... because `isMinorValid` really means versions were not good at the patch level but are good at the minor level.